### PR TITLE
Partitioned `dot`

### PR DESCRIPTION
**Pull request by paraynaud**
#23

### CODE REVIEW

File: src/linearAlgebra.jl [hunk 1]
```diff
@@ -8,7 +8,28 @@ function norm(pv::PartitionedVector, p::Real = 2; kwargs...)
...
+  acc = 0
+  for i in 1:N
+    eev1 = pv1[i].vec
+    eev2 = pv2[i].vec
+    acc += dot(eev1, eev2)
+  end
+  return acc
+end
+
+dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}, ::Val{false}, ::Val{true}; kwargs...) where {T} = dot(pv2, pv1)
```
Comment (tmigot) [inline src/linearAlgebra.jl:24]: Maybe that would be slightly easier to read? But agree that what you have should work.
```suggestion
dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}, ::Val{false}, ::Val{true}; kwargs...) where {T} = dot(pv2, pv1, Val(true), Val(false))
```

File: src/linearAlgebra.jl [hunk 2]
```diff
@@ -8,7 +8,28 @@ function norm(pv::PartitionedVector, p::Real = 2; kwargs...)
   return _norm
 end
 
-function dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}; kwargs...) where {T}
+dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}; kwargs...) where {T} = dot(pv1, pv2, Val(pv1.simulate_vector), Val(pv2.simulate_vector); kwargs...)
+
+function dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}, ::Val{true}, ::Val{false}; kwargs...) where {T}
```
Comment (tmigot) [inline src/linearAlgebra.jl:13]: I think it's worth adding a short docstring here. I am giving an example, that would just miss a sentence on the `::Val{true}, ::Val{false}` thing.

```suggestion
"""
    dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}, ::Val{true}, ::Val{false}; kwargs...)

Compute the dot product between two partitioned vectors.
"""
function dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}, ::Val{true}, ::Val{false}; kwargs...) where {T}
```

**Review by tmigot — COMMENTED** [review]
Looks great, I have just two short comments. Let me know what you think.

File: src/linearAlgebra.jl (hunk 1 above)
Comment (paraynaud) [reply src/linearAlgebra.jl:24]: You're right, it will be easier to read.

**Review by tmigot — APPROVED** [review]
Thanks!

**Pull request merged**